src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/DefaultRazorTagHelperContextDiscoveryPhase.cs

```diff
@@ -69,6 +69,8 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
         var context = TagHelperDocumentContext.Create(tagHelperPrefix, matches.ToImmutableArray());
         codeDocument.SetTagHelperContext(context);
         codeDocument.SetPreTagHelperSyntaxTree(syntaxTree);
+
+        visitor.Dispose();
```

Comment:
Consider `using (visitor) {` on line 55-ish, and ending it here, so its clearer to everyone that its being disposed, and so that there is a try..finally in place

Reply:
I can change to that, I didn't go that route initially as it didn't fall directly into the using pattern and it's really not a big deal if the disposal doesn't end up happening.

Reply:
> and it's really not a big deal if the disposal doesn't end up happening

Is it not? I would have thought that would mean the pool allocates lots of new builders, defeating the purpose of the PR.

Reply:
Let me rephrase, it's not a big deal if it's an exceptional case where this occurs, not if it's the normal code path (which would probably indicate bigger problems than avoiding these allocations)